### PR TITLE
feat(core): add PUT /configs/jwt-customizer API

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add `PUT /configs/jwt-customizer/:keyType` API

## Testing

Tested locally via Postman, also covered by unit test and integration tests.

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/logto-config.ts [hunk 1]
```diff
@@ -182,4 +197,57 @@ export default function logtoConfigRoutes<T extends AuthedRouter>(
...
+    async (ctx, next) => {
+      const {
+        params: { tokenType },
+        body,
+      } = ctx.guard;
+
+      // Manually implement the request body type check, the flow aligns with the actual `koaGuard()`.
+      // Use ternary operator to get the specific guard brings difficulties to type inference.
+      if (tokenType === LogtoJwtTokenKeyType.AccessToken) {
+        tryParse('body', jwtCustomizerAccessTokenGuard, body);
```
Comment (simeng-li) [inline packages/core/src/routes/logto-config.ts:225]: XXXX = jwtCustomizerAccessTokenGuard.safeParse();

File: packages/core/src/queries/logto-config.ts [hunk 2]
```diff
@@ -47,11 +52,28 @@ export const createLogtoConfigQueries = (pool: CommonQueryMethods) => {
       returning *
     `);
 
+  // Can not narrow down the type of value if we utilize `buildInsertIntoWithPool` method.
+  const insertOrUpdateJwtCustomizer = async <T extends LogtoJwtTokenKey>(
```
Comment (simeng-li) [inline packages/core/src/queries/logto-config.ts:56]: nit: upsertJwtCustomizer

File: packages/core/src/routes/logto-config.ts [hunk 3]
```diff
@@ -29,6 +37,21 @@ const getOidcConfigKeyDatabaseColumnName = (key: LogtoOidcConfigKeyType): LogtoO
     ? LogtoOidcConfigKey.PrivateKeys
     : LogtoOidcConfigKey.CookieKeys;
 
+const getLogtoJwtTokenKey = (key: LogtoJwtTokenPath): LogtoJwtTokenKey =>
```
Comment (simeng-li) [inline packages/core/src/routes/logto-config.ts:40]: nit:
|```ts
const jwtTokenPathConfig = {
[LogtoJwtTokenPath.AccessToken]: {
  tokenKey: LogtoJwtTokenKey.AccessToken
  guard: jwtCustomizerAccessTokenGuard
},
[LogtoJwtTokenPath.ClientCredentials]: {
  tokenKey: LogtoJwtTokenKey.ClientCredentials
  guard: jwtCustomizerClientCredentialsGuard
}

const {tokenKey, guard} = jwtTokenPathConfig[tokenTypePath];
const payload = parse('body', jwtCustomizerAccessTokenGuard, body)
}
```

File: packages/core/src/routes/logto-config.ts [hunk 4]
```diff
@@ -182,4 +210,42 @@ export default function logtoConfigRoutes<T extends AuthedRouter>(
...
+        tokenTypePath: z.nativeEnum(LogtoJwtTokenPath),
+      }),
+      /**
+       * Use `z.unknown()` to guard the request body as a JSON object, since the actual guard depends
+       * on the `tokenTypePath` and we can not get the value of `tokenTypePath` before parsing the request body,
+       * we will do more specific guard as long as we can get the value of `tokenTypePath`.
+       *
+       * Should specify `body` in koaGuard, otherwise the request body is not accessible even via `ctx.request.body`.
+       */
+      body: z.unknown(),
```
Comment (simeng-li) [inline packages/core/src/routes/logto-config.ts:227]: can remove these?

Comment (darcyYe) [reply packages/core/src/routes/logto-config.ts:225]: We can't, see the comments. If we do not explicitly define `body` guard, we can not access `body` data in the following API operation handler.

Comment (simeng-li) [reply packages/core/src/routes/logto-config.ts:220]: use koaBody.

Comment (darcyYe) [reply packages/core/src/routes/logto-config.ts]: updated

Comment (darcyYe) [reply packages/core/src/routes/logto-config.ts]: use `koaBody()` brings trouble to the swagger file parser. Keep it as it is?
image

**Review by simeng-li — APPROVED** [review]

File: packages/core/src/routes/logto-config.ts (hunk 4 above)
Comment (darcyYe) [reply packages/core/src/routes/logto-config.ts:220]: See LOG-8376 to fix this later.

**Pull request merged**